lib/builder.ts

```diff
@@ -166,17 +171,35 @@ class Builder extends EventEmitter {
     // cleanup, or restarting the build itself.
     this._cancelationRequest = new CancelationRequest(pipeline);
 
+    let skipFinally = false;
     try {
       await pipeline;
       this.buildHeimdallTree(this.outputNodeWrapper);
+    } catch (e) {
+      if (RetryCancellationRequest.isRetry(e)) {
+        this._cancelationRequest = undefined;
+        await new Promise((resolve, reject) => {
+          setTimeout(() => {
+            try {
+              resolve(this.build());
+            } catch(e) {
+              reject(e);
+            }
+          }, e.retryIn);
```

Comment:
Verify the new approach to use cancellation/retries to implement the debounce works

Reply:
consider `retryIn` -> `debounce` ?